src/System.Text.Json/src/System/Text/Json/Node/JsonObject.cs

```diff
@@ -394,7 +181,46 @@ public void Add(string propertyName, IEnumerable<JsonNode> propertyValues)
         /// <exception cref="ArgumentNullException">
         ///   Provided property name is null.
         /// </exception>
-        public bool Remove(string propertyName) => propertyName != null ? _dictionary.Remove(propertyName) : throw new ArgumentNullException(nameof(propertyName));
+        public bool Remove(string propertyName)
+        {
+            if (propertyName == null)
+            {
+                throw new ArgumentNullException(nameof(propertyName));
+            }
+
+            if (!_dictionary.ContainsKey(propertyName))
```

Comment:
This method is currently "tripple-dipping" the dictionary.  ContainsKey, get_Item, and Remove.

They can all be combined into

```C#
if (!_dictionary.Remove(propertyName, out JsonObjectProperty value))
```

If it returns false, there was no match.  If it returns true, there's now no match and you have the existing value to manipulate the linked list.

Reply:
Thanks, I wasn't aware of this method! And I am rewriting it anyway a little bit now, because of adding `StringComparison`.

Reply:
So this `Remove` does not work for netfx :(